src/sentry/static/sentry/app/views/performance/traceDetails/content.tsx

```diff
@@ -143,20 +128,45 @@ class TraceDetailsContent extends React.Component<Props, State> {
       )
     );
 
+    const queryLength = searchQuery.length;
+
     const fuse = await createFuzzySearch(transformed, {
-      keys: ['indexed', 'tagKeys', 'tagValues', 'dataKeys', 'dataValues'],
-      includeMatches: false,
+      keys: ['indexed'],
+      includeMatches: true,
       threshold: 0.6,
       location: 0,
       distance: 100,
       maxPatternLength: 32,
     });
 
-    const results = fuse.search<FuseResult>(searchQuery);
-    const matched = results.map(result => result.item.transaction);
+    const fuseMatches = fuse
+      .search<IndexedFusedTransaction>(searchQuery)
+      /**
+       * Sometimes, there can be matches that don't include any
+       * indices. These matches are often noise, so exclude them.
+       */
+      .filter(({matches}) => matches.length)
+      .map(({item}) => item.transaction.event_id);
+
+    /**
+     * We match on the id as a substring match. We only consider ids
+     * that are a length that is a multiple of 8
+     */
+    const idMatches =
+      queryLength % 8 === 0
```

Comment:
I don't quite get this check for multiples? You can only find id's if you type multiples of 8?

Reply:
The idea is to only search ids when the input resembles an id. Though it might be a better experience if we just always checked it. @wmak thoughts on removing this length check?

Reply:
If I recall this search can be slow which is why I suggested onlyd doing this extra work at multiples of 8
8 == the short ids we show in the UI if someone copy pastes
16 == the length of span ids
32 == the length of actual uuids

Reply:
I did some naive benchmarking on this. And found doing a substring match on the 100 transaction ops and transaction names takes <1ms in my browser. Given that result, I think it's safe to say there is no need to be concerned with this being too slow.

If there are no other reasons, I think we should just remove this check.

Reply:
Good to see benchmarks :+1: i thought the fuzzy search was a lot slower than that for some reason

Reply:
You are correct in that fuzzy search can be a lot slower in some cases (we've disabled some of these options to get a reasonably performant search). However, this specific case is doing a substring match on the ids which is limited to 100 transactions which is significantly faster.